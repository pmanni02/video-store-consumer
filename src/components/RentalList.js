import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Movie from './Movie';
import './RentalList.css'
import axios from 'axios';

class RentalList extends Component {
  constructor() {
    super();
    this.state = {
      movies: [],
    };
  }

  componentDidMount = () => {
    let query =  'http://localhost:3000/movies'
    console.log(query)

    axios.get(query)
      .then((response) => {
        console.log(response);
        this.setState({
          movies: response.data
        })
      })
      .catch((error) => {
        this.setState({
          error: error.message
        });
        if (error.message) {
          this.renderError();
          console.log(error.message);
        }
      }
    );
  }

  renderMovieList = () => {
    const movieList = this.state.movies.map((item, index) => {
      return(
        <Movie
          key={index}
          id={item.id}
          title={item.title}
          overview={item.overview}
          release={item.release_date}
          image_url={item.image_url}
          movieCallback={this.pickMovie}
        />
      )
    })
    return movieList
  }

  renderError = () => {
    return (
      <p className='validation-errors-dislay'>
        {this.state.error}
      </p>
    )
  }

  pickMovie = (id) => {
    this.props.pickMovieDetailCallback(id)
  }

  render() {
    return (
        <div className="rental-list">
          {this.renderMovieList()}
        </div>
    )
  }
}

export default RentalList;

RentalList.propTypes = {
  pickMovieDetailCallback: PropTypes.func,
}
