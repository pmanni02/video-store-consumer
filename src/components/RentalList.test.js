import React from 'react';
import { shallow } from 'enzyme';
import RentalList from './Movie';

describe('RentalList', () => {
  test('snapshot', () => {
    const wrapper = shallow(
      <RentalList
        pickMovieDetailCallback={()=>{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
