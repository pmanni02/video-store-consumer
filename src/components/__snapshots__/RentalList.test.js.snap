// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RentalList snapshot 1`] = `
<article
  className="movie"
>
  <h3 />
  <img
    onClick={[Function]}
    title="Click to select"
  />
  <p />
  <p />
</article>
`;
